src/DistortableCollection.js

```diff
@@ -199,19 +199,65 @@ L.DistortableCollection = L.FeatureGroup.extend({
     return reduce / imgs.length;
   },
 
-  isJsonDetected(currentURL) {
+  _extractJSONDownloadURL(currentURL) {
+    const startIndex = currentURL.lastIndexOf('=');
+    const jsonDownloadURL = currentURL.slice(startIndex + 1);
+
+    return jsonDownloadURL;
+  },
+
+  _isJsonDetected(currentURL) {
     if (currentURL.includes('?json=')) {
-      startIndex = currentURL.lastIndexOf('.');
-      fileExtension = currentURL.slice(startIndex + 1);
+      const startIndex = currentURL.lastIndexOf('.');
+      const fileExtension = currentURL.slice(startIndex + 1);
 
       if (fileExtension === 'json') {
-        console.log('JSON found in map shareable link');
+        console.log('JSON found in map shareable link'); // left here purposely
         return true;
       }
     }
+
     return false;
   },
 
+  async _generateFromJson(jsonDownloadURL) {
```

Comment:
```suggestion
  async _recreateMapFromJson(json) {
```

How about this for a name?

And, what do you think of separating the remote fetching from the regenerating, because we could then use the regeneration code on JSON from another source, like localStorage? So we could have:

`_recreateMapFromJson(json)`

and

`_recreateMapFromJsonUrl(url)` which just fetches the URL and runs `_recreateMapFromJson(json)` on the response?

Reply:
Okay, sounds good. Thanks!